### PR TITLE
Give extending classes access to some of the data grid's paint utilities.

### LINKED ISSUE #110: Give extending classes access to some of the data grid's paint utilities.
Hi, my name is Logan McNichols. I am an intern for Jupyter Cal Poly. I am working on a tabular data editor powered by the `DataGrid` along with @kgoo124 and @ryuntalan.

We are working on a design that involves painting `LabIcons` onto the `DataGrid`'s canvas to correspond the the type of field in a given column. To achieve this, we would like to be able to extend the `DataGrid` class to add custom drawing functions. To do this, our class would need to be able to access some properties & methods of the `DataGrid` that are currently set to `private`.

The methods we would need access to are
`_paintContent`
`_repaintContent` 
& `_repaintOverlay`

The properties we would need getters for are
- `_canvasGC`
- `_rowSections`
- `_columnSections`
- `_columnHeaderSections`
- `_rowHeaderSections`.

### PR DESCRIPTION
**Pull request by lmcnichols**
Hi, my name is Logan McNichols. I am an intern for Jupyter Cal Poly. I am working on a tabular data editor powered by the `DataGrid` along with @kgoo124 and @ryuntalan.

We are working on a design that involves painting `LabIcons` onto the `DataGrid`'s canvas to correspond the the type of field in a given column. To achieve this, we would like to provide getters for some of the `DataGrid`'s properties and change some `private` methods to `protected`.

This PR would change the access modifier from `private` to `protected` on the following methods.
- `paintContent`
`repaintContent` 
& `repaintOverlay`

It would also add getters for the following properties.
- `_canvasGC`
- `_rowSections`
- `_columnSections`
- `_columnHeaderSections`
- `_rowHeaderSections`.

Fixes #110

### CODE REVIEW

File: packages/datagrid/src/datagrid.ts [hunk 1]
```diff
@@ -728,6 +728,41 @@ class DataGrid extends Widget {
       this.dataModel instanceof MutableDataModel;
   }
 
+  /**
+   * The renduring context for painting the data grid.
+   */
+  get canvasGC(): CanvasRenderingContext2D {
```
Comment (blink1073) [inline packages/datagrid/src/datagrid.ts:734]: Do these need to be `public`?  Would `protected` getters work for your use case?

File: packages/datagrid/src/datagrid.ts [hunk 2]
```diff
@@ -728,6 +728,41 @@ class DataGrid extends Widget {
       this.dataModel instanceof MutableDataModel;
   }
 
+  /**
+   * The renduring context for painting the data grid.
```
Comment (blink1073) [inline packages/datagrid/src/datagrid.ts:732]: (sp) rendereing

Comment (lmcnichols) [reply packages/datagrid/src/datagrid.ts]: Fixed, thanks!

File: packages/datagrid/src/datagrid.ts (hunk 1 above)
Comment (lmcnichols) [reply packages/datagrid/src/datagrid.ts]: Yes, `protected` is fine.

**Comment by nmichaud** [comment]
I'm not sure this is necessary - it's possible to leverage a combination of metadata and renderers to draw images/icons onto the grid. What are you trying to do in a subclass?

**Comment by kgoo124** [comment]
@nmichaud We are creating a ghost row/column for the data grid to add new rows/columns. Here is a [prototype](https://github.com/jupytercalpoly/jupyterlab-tabular-data-editor/issues/113) of what we created.

**Comment by nmichaud** [comment]
@kgoo124 ah ok, that looks interesting! Definitely not possible with the current grid code. It looks like you want to draw on the canvas outside of the grid region.

**Comment by kgoo124** [comment]
@nmichaud yes that's right. Since the ghost rows/columns serve as UI to add a new row/column, we want to be able to paint an extra row/column on the grid without having the manage the data model underneath it

**Pull request merged**